deployment/bicep/modules/acr.bicep

```diff
@@ -33,23 +30,12 @@ resource acr 'Microsoft.ContainerRegistry/registries@2021-06-01-preview' = {
   }
 }
 
-resource assignAcrPullToAks1 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
-  name: guid(resourceGroup().id, acrName, aks1PrincipalId, 'AssignAcrPullToAks')
-  scope: acr
-  properties: {
-    description: 'Assign AcrPull role to AKS #1'
-    principalId: aks1PrincipalId
-    principalType: 'ServicePrincipal'
-    roleDefinitionId: '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/${roleAcrPull}'
-  }
-}
-
-resource assignAcrPullToAks2 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
-  name: guid(resourceGroup().id, acrName, aks2PrincipalId, 'AssignAcrPullToAks')
+resource assignAcrPullToAks 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  name: guid(resourceGroup().id, acrName, aksPrincipalId, 'AssignAcrPullToAks')
   scope: acr
   properties: {
-    description: 'Assign AcrPull role to AKS #2'
-    principalId: aks2PrincipalId
+    description: 'Assign AcrPull role to AKS provided (level 3)'
```

Comment:
We could update the description to keep it generic, i.e. the ACR bicep module should not be aware of which level/AKS it will be catering to, that's defined by the higher level abstractions/scripts?

Reply:
Agree, and might be cleaner together with the comment above. Would be cleaner to separate role assignment to new .bicep file. Wdyt?